dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pPanel.vue

```diff
@@ -0,0 +1,332 @@
+            <div v-if="!isExpanded(index)">
+              <span
+                :class="`p-badge badge-${colorOfCategory(arrayOfKpiDataObject[0])}`"
+                :data-test="arrayOfKpiDataObject[0]"
+                >{{ arrayOfKpiDataObject[0].toUpperCase() }}</span
+              >
+              <button class="pt-1 pr-3 d-cursor-pointer d-chevron-style">
+                <span class="pr-1 pt-1 pi pi-chevron-right d-chevron-font"></span>
+              </button>
+            </div>
+            <div v-if="isExpanded(index)">
+              <span :class="`p-badge badge-${colorOfCategory(arrayOfKpiDataObject[0])}`">{{
+                arrayOfKpiDataObject[0].toUpperCase()
+              }}</span>
+              <button class="pt-2 pr-3 d-cursor-pointer d-chevron-style">
+                <span class="pr-1 pi pi-chevron-down d-chevron-font"></span>
+              </button>
+            </div>
+          </div>
+        </div>
+        <div v-show="isExpanded(index)">
+          <DisplayFrameworkDataTable
+            :arrayOfKpiDataObjects="arrayOfKpiDataObject[1]"
+            :list-of-reporting-periods-with-data-id="listOfDataSetReportingPeriods"
+            headerInputStyle="display: none;"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { KpiDataObject, KpiValue } from "@/components/resources/frameworkDataSearch/KpiDataObject";
+import { PanelProps } from "@/components/resources/frameworkDataSearch/PanelComponentOptions";
+import DisplayFrameworkDataTable from "@/components/resources/frameworkDataSearch/DisplayFrameworkDataTable.vue";
+import { p2pDataModel } from "@/components/resources/frameworkDataSearch/p2p/P2pDataModel";
+import { ApiClientProvider } from "@/services/ApiClients";
+import { ReportingPeriodOfDataSetWithId, sortReportingPeriodsToDisplayAsColumns } from "@/utils/DataTableDisplay";
+import { Category, Subcategory } from "@/utils/GenericFrameworkTypes";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import { reformatValueForDisplay } from "@/utils/FrameworkPanelDisplay";
+import { DataAndMetaInformationPathwaysToParisData } from "@clients/backend";
+import Keycloak from "keycloak-js";
+import { defineComponent, inject } from "vue";
+import Column from "primevue/column";
+import DataTable from "primevue/datatable";
+
+export default defineComponent({
+  name: "P2pPanel",
+  components: { DisplayFrameworkDataTable, DataTable, Column },
+  data() {
+    return {
+      firstRender: true,
+      waitingForData: true,
+      resultKpiData: null as KpiDataObject,
+      p2pDataAndMetaInfo: [] as Array<DataAndMetaInformationPathwaysToParisData>,
+      listOfDataSetReportingPeriods: [] as Array<ReportingPeriodOfDataSetWithId>,
+      mapOfKpiKeysToDataObjects: new Map() as Map<string, KpiDataObject>,
+      mapOfKpiKeysToDataObjectsArrays: new Map() as Map<string, Array<KpiDataObject>>,
```

Comment:
it is rather a `mapOfCategoryKeysToDataObjectArrays`.

Reply:
Changed